### PR TITLE
Add ssn to metadata 7959a  7959f1 7959f2

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary
As a PEGA user, I need SSN or x-amz-meta-ssn_or_tin for all forms to process in the PEGA workflow.
AC: x-amz-meta-ssn_or_tin is in the metadata for all IVC CHAMPVA forms.

## Related issue(s)
Similar work  for 7959C
(https://github.com/department-of-veterans-affairs/vets-api/pull/17578/files)

## Testing done
Successfully ran unit test for 7959a, 7959f1 and 7959f2.
No pdfs were generated

### CODE REVIEW

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def metadata
         'source' => 'VA Platform Digital Forms',
         'docType' => @data['form_number'],
         'businessLine' => 'CMP',
+        'ssn_or_tin' => @data.dig('veteran', 'ssn_or_tin'),
```
Comment (michaelclement) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:25]: This line will have to be
`'ssn_or_tin' => @data.dig('applicant_member_number'),`

File: modules/ivc_champva/spec/models/vha_10_7959a_spec.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@
       'veteran' => {
         'full_name' => { 'first' => 'John', 'middle' => 'P', 'last' => 'Doe' },
         'va_claim_number' => '123456789',
+        'ssn_or_tin' => '123456789',
         'address' => { 'country' => 'USA', 'postal_code' => '12345' }
       },
```
Comment (michaelclement) [inline modules/ivc_champva/spec/models/vha_10_7959a_spec.rb:20]: There is no `veteran` block in the data coming from 10-7959a. We'll want this section to be:

```javascript
{
  'applicant_name' => { 'first' => 'John', 'middle' => 'P', 'last' => 'Doe' },
  'applicant_member_number' => '123456789',
  'applicant_address' => { 'country' => 'USA', 'postal_code' => '12345' },
}
```

**Review by michaelclement — DISMISSED** [review]

File: modules/ivc_champva/spec/models/vha_10_7959a_spec.rb (hunk 2 above)
Comment (cloudmagic80) [reply modules/ivc_champva/spec/models/vha_10_7959a_spec.rb]: I thought you were going to change it on your commit

File: modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb [hunk 3]
```diff
@@ -16,12 +16,13 @@ def initialize(data)
...
-        'veteranFirstName' => @data.dig('veteran', 'full_name', 'first'),
-        'veteranLastName' => @data.dig('veteran', 'full_name', 'last'),
-        'zipCode' => @data.dig('veteran', 'address', 'postal_code'),
+        'veteranFirstName' => @data.dig('applicant_name', 'first'),
+        'veteranLastName' => @data.dig('applicant_name', 'last'),
+        'zipCode' => @data.dig('applicant_address', 'postal_code'),
         'source' => 'VA Platform Digital Forms',
         'docType' => @data['form_number'],
         'businessLine' => 'CMP',
+        'ssn_or_tin' => @data['applicant_member_number'],
```
Comment (rmtolmach) [inline modules/ivc_champva/app/models/ivc_champva/vha_10_7959a.rb:25]: This metadata isn't logged anywhere, is it?

**Review by michaelclement — APPROVED** [review]

**Pull request merged**